plugins/analysis-icu/src/main/java/org/elasticsearch/index/mapper/ICUCollationFieldMapper.java

```
@@ -0,0 +1,623 @@
+import org.elasticsearch.search.DocValueFormat;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.LongSupplier;
+
+public class ICUCollationFieldMapper extends FieldMapper {
+
+    public static final String CONTENT_TYPE = "icu_collation";
+
+    public static class Defaults {
+        public static final MappedFieldType FIELD_TYPE = new CollationFieldType();
+    }
+
+    public static final class CollationFieldType extends StringFieldType {
+        public CollationFieldType() {
+            setStored(false);
+            setTokenized(false);
+            setIndexOptions(IndexOptions.NONE);
+            setHasDocValues(true);
+            setDocValuesType(DocValuesType.SORTED);
+            freeze();
+        }
+
+        protected CollationFieldType(CollationFieldType ref) {
+            super(ref);
+        }
+
+        public CollationFieldType clone() {
+            return new CollationFieldType(this);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder() {
+            failIfNoDocValues();
+            return new DocValuesIndexFieldData.Builder();
+        }
+
+        public static DocValueFormat COLLATE_FORMAT = new DocValueFormat() {
+            @Override
+            public String getWriteableName() {
+                return "collate";
+            }
+
+            @Override
+            public void writeTo(StreamOutput out) throws IOException {
+            }
+
+            @Override
+            public String format(long value) {
+                return Long.toString(value);
+            }
+
+            @Override
+            public String format(double value) {
+                return Double.toString(value);
+            }
```

format on long and double values should just throw an exception